src/units/abilities.cpp

```diff
@@ -523,6 +569,46 @@ bool unit::ability_affects_adjacent(const std::string& ability, const config& cf
 	return false;
 }
 
+bool unit::ability_affects_distant(const std::string& ability, const config& cfg, const map_location& loc, const unit& from, const map_location& from_loc) const
+{
+	unsigned int radius = 0;
+	bool illuminates = ability == "illuminates";
```

Comment:
It looks like this isn't new - `ability_affects_adjacent()` does the same thing already - but why is illuminates handled differently than any other ability here?

Reply:
Same logic what for ability_affects_adjacent() and i don't know why, perhaps because illuminates affect adjacent hexes when appied to self

Reply:
It's probably because filters incorporate the time of day affecting a hex ([filter_location]), and illuminates modifies this Tod that the processing of 'illuminates' abilities is different from the rest